### ISSUE TITLE
Error: Assertion failed: (vMsgsBeforeOpenLog), function LogPrintStr, file util.cpp, line 348 on startup.

### ISSUE THREAD

**Issue by yellowred** [bitcoin/bitcoin#12478]
Hi,

I am trying to run a full node on Macbook Pro with OS `Mac OSX v10.13.3` with freshly download bitcoin-qt `Bitcoin Core version v0.15.1 (64-bit)`

However it fails with:
```
➜  MacOS ./Bitcoin-Qt
Assertion failed: (vMsgsBeforeOpenLog), function LogPrintStr, file util.cpp, line 348.
[1]    948 abort      ./Bitcoin-Qt
```

I expect it to launch successfully and show an invitation to choose the datadir.

The computer specs:
```
 Model Name:	MacBook Pro
  Model Identifier:	MacBookPro14,3
  Processor Name:	Intel Core i7
  Processor Speed:	2.8 GHz
  Number of Processors:	1
  Total Number of Cores:	4
  L2 Cache (per Core):	256 KB
  L3 Cache:	6 MB
  Memory:	16 GB
  Boot ROM Version:	MBP143.0169.B00
  SMC Version (system):	2.45f0
```

The error details are in the attachment.
[error.txt](https://github.com/bitcoin/bitcoin/files/1734513/error.txt)

EDIT1: I am able to compile src and launch bitcoind successfully:
```
➜  bitcoin git:(master) ✗ ./src/bitcoind -version
Bitcoin Core Daemon version v0.16.99.0-8a98dfeeb
```

**Comment by fanquake** [comment]
@yellowred  Are you still having the same issue with 0.16.0?

**Comment by yellowred** [comment]
I have updated to Bitcoin Core 0.16.0 release pre-compiled and packed by the core team and it works perfectly. I had issues with the version that I have compiled myself regarding networking and I don't have such issues with the lates binary anymore.

**Comment by jnewbery** [comment]
ok, sounds like this is fixed in V0.16. Closing.

We can reopen if you think this issue still isn't fully resolved.
